src/server/handlers/store.js

```diff
@@ -1,23 +1,35 @@
 import 'isomorphic-fetch';
 
+import db from '../db';
 import { conf } from '../helpers/config';
 
 export const registerName = async (req, res) => {
   const snapName = req.body.snap_name;
   const root = req.body.root;
   const discharge = req.body.discharge;
 
-  const response = await fetch(`${conf.get('STORE_API_URL')}/register-name/`, {
-    method: 'POST',
-    headers: {
-      'Authorization': `Macaroon root="${root}", discharge="${discharge}"`,
-      'Content-Type': 'application/json',
-      'Accept': 'application/json'
-    },
-    body: JSON.stringify({ snap_name: snapName })
+  await db.transaction(async (trx) => {
+    // XXX cjwatson 2017-03-17: This may need to move somewhere else in
```

Comment:
Maybe it can be moved to `authorizeSnap` in `handlers/launchpad.js` as it seems we call authorize directly after registering name.